section3/reflection.md

```diff
@@ -2,12 +2,24 @@
 
 1. In Section 3, we removed even more supports and pushed you to learn a bit more independently, read through technical documentation, and use your resources more. How did that feel?
 
+* Great! The issue with learning on my own was not knowing where to go next. This format works great for giving you everything you need to succeed, without holding your hand the entire time.
+
 2. Think of a web or mobile application you use regularly. Where is an example of a place that the developers probably use Arrays? Explain.
 
+* Arrays are used anywhere a list need be stored. This could be an array of all active users on Twitch, an array of dogs over 5 years old on an adoption site, or an array of bookmarks on Google Chrome.
+
 3. In your own words, explain what an Array method is.
 
+* An array method is the method used to create an array. From the book, there are two. One of them is the array literal which is preferred and how we have done arrays in this course, and the other is the array constructor which looks much worse anyway.
+
 4. What is the purpose of a loop? When we would use a loop in conjunction with array vs. without?
 
+* A purpose of a loop is to perform multiple tasks over and over again. A loop with an array would be used to manipulate the items in that array. An array is not always necessary though, and a loop can be used with even one lone variable.
+
 5. If you had to teach a 5 year old about loops, how would you explain it?
 
+* Let's say I wanted to do something a bunch of times, like counting! Now you and I could stand here and count to one million together, but that would probably take forever right? It's way easier to just tell a computer to count to one million for us. If we got on the computer and started typing out every individual number to one million, once again we'd take way too long! What we can actually do here is take a couple seconds and give the computer only a couple lines of instructions, and it will count to one million, every digit, in the blink of an eye for us, and it won't even complain! Pretty cool huh?
+
 6. What questions do you still have about Arrays and loops?
+
+* It's getting to the point where I'm not sure what to ask. One question I could pose is using loops with an array of different data types. What we have done so far is arrays with one type, if you were to write an array with a boolean(not sure what I'm talking about), a number, and a string, and then wrote a for-loop, how would that even work? Would the datatypes be coerced like in the use of `==`?
```

Comment:
Ha, Great question!!

It's generally a best practice to not mix data types within an Array for just this reason - "how would that even work"? We'd probably run into some hairy issues or not be able to use our data. JavaScript is sometimes called "the wild wild west" because it lets us do horrible things like that, even though we probably should not.

If you're curious to learn a bit more before Mod 1 - look into Object Literals in JavaScript. It's another collection data type that lets you mix data types and not have the same struggles.